### PR TITLE
Revert passing $request to renderWithExtraParam

### PR DESCRIPTION
**Pull request by VincentLanglet**
Follow https://github.com/sonata-project/SonataAdminBundle/pull/7238#issuecomment-856777962

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Multiples things here:
- Should we remove `addRenderExtraParams` and add directly the two lines inside the `renderWithExtraParams` method ?
- Should we add final to `renderWithExtraParams/addRenderExtraParams` ?
This way if we need one day to add an extra param to use it in the twig, it won't be a BC break.

Shouldn't we add more `final` to others method ? Some of the following method:
- renderJson
- isXmlHttpRequest
- isPreviewRequested, isPreviewApproved, isInPreviewMode, isPreviewDeclined ?
- getAclUsers, getAclRoles ?
- validateCsrfToken, getCsrfToken ?
- escapeHtml

**Comment by jordisala1991** [comment]
the add render extra parameters works for other Sonata bundles like media: https://github.com/sonata-project/SonataMediaBundle/blob/master/src/Controller/GalleryAdminController.php#L23-L29

**Comment by VincentLanglet** [comment]
> Edit2: It can be used to pass more parameters to all CRUDController methods without having to extend all of them to... (not sure if we want to promote such things)

That's the main question I had in mind. Since we plan to move the base_template into a service, the method will only add the admin. IMHO, it could be nice to deprecate the `renderWithExtraParams` and directly add the admin param to all the action.

If we plan to do this, we should start adding `final` to this method in order to promote others way to pass parameters to the view.

**Review by jordisala1991 — DISMISSED** [review]

File: UPGRADE-4.0.md [hunk 1]
```diff
@@ -184,6 +184,10 @@ When there is no searchable filters, `SearchHandler::search()` returns `null`. P
 ## Sonata\AdminBundle\Controller\CRUDController
 When the service `security.csrf.token_manager` is not available, `getCsrfToken()` returns `null`. Previously, it was returning `false`.
 
+The `isXmlHttpRequest()`, `redirectTo()`, `isPreviewApproved()`, `isInPreviewMode()`, `isPreviewDeclined()`,
+`validateCsrfToken()` signatures was changed. They now require to pass the request as first argument.
```
Comment (phansys) [inline UPGRADE-4.0.md:188]:
```suggestion
`validateCsrfToken()` signatures were changed. They now require to pass the request as first argument.
```

**Review by phansys — APPROVED** [review]

**Comment by VincentLanglet** [comment]
> Multiples things here:
> 
> * Should we remove `addRenderExtraParams` and add directly the two lines inside the `renderWithExtraParams` method ?
> * Should we add final to `renderWithExtraParams/addRenderExtraParams` ?
>   This way if we need one day to add an extra param to use it in the twig, it won't be a BC break.
> 
> Shouldn't we add more `final` to others method ? Some of the following method:
> 
> * renderJson
> * isXmlHttpRequest
> * isPreviewRequested, isPreviewApproved, isInPreviewMode, isPreviewDeclined ?
> * getAclUsers, getAclRoles ?
> * validateCsrfToken, getCsrfToken ?
> * escapeHtml

I merged this since it was approved and this will help @jordisala1991 on media bundle.

But we can still continue the discussion about which method should be final.
I'd like you opinion @sonata-project/contributors

**Pull request merged**